contract/tests/test_contract.py

```diff
@@ -103,6 +104,16 @@ def test_contract_yearly(self):
         self.assertEqual(self.contract_yearly.recurring_next_date,
                          new_date.strftime('%Y-%m-%d'))
 
+    def test_contract_monthly_lastday(self):
+        self.contract_monthly_lastday.recurring_create_invoice()
+        invoices_monthly_lastday = self.env['account.invoice'].search(
+            [('contract_id', '=', self.contract_monthly_lastday.id)])
+        self.assertTrue(invoices_monthly_lastday)
+        new_date = self.current_date + relativedelta(
```

Comment:
Don't make the same compute in the test. In order to test features, you need to hardcode data. Imagine that the implementation of relativedelta changes, and now this return another date. With this test, you won't detect it, because both uses the same method for compute. The main thing in tests is to compare some fixed data (starting date, interval... and the resulting date) with the data obtained by your methods.